test/check-argo-status.sh

```diff
@@ -15,38 +15,37 @@
 # limitations under the License.
 
 
-echo "check status of argo workflow $ARGO_WORKFLOW...."
-# probing the argo workflow status until it completed. Timeout after 30 minutes
-for i in $(seq 1 ${PULL_ARGO_WORKFLOW_STATUS_MAX_ATTEMPT})
-do
-  WORKFLOW_STATUS=`kubectl get workflow $ARGO_WORKFLOW -n ${NAMESPACE} --show-labels`
-  echo $WORKFLOW_STATUS | grep ${WORKFLOW_COMPLETE_KEYWORD} && s=0 && break || s=$? && printf "Workflow ${ARGO_WORKFLOW} is not finished.\n${WORKFLOW_STATUS}\nSleep for 20 seconds...\n" && sleep 20
+#Wait for the workflow to appear
+while [ "$(kubectl get workflow/$ARGO_WORKFLOW --namespace "$NAMESPACE" -o=jsonpath='{.status.phase}')" == "" ]; do
```

Comment:
looks like it can wait forever. it can hold gke resource and never release.

Reply:
Can the workflow never appear in Kubernetes?

Reply:
yes. this can also wait forever if it appears but never finish.

Reply:
Prow has 2h timeout. Do you think it's ok to rely on it?

Reply:
would that kill our test cluster or just terminate the pod in prow cluster?